packages/flutter/lib/src/material/snack_bar_theme.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Configures whether [SnackBar] should be fixed to the bottom or floating
+/// above other widgets fixed to the bottom, such as [FloatingActionButton].
+enum SnackBarBehavior {
+  /// Fixed the [SnackBar] position to the bottom of the [Scaffold] when
+  /// possible. One of the possible scenarios where [SnackBar] will be shown
+  /// above another widget is above a [BottomNavigationBar]. Other than that
+  /// [SnackBar] will cause other non-fixed widgets inside [Scaffold] to be
+  /// pushed above (e.g. [FloatingActionButton]).
+  fixed,
+
+  /// Change the design and behavior of [SnackBar] to float as described in
+  /// <https://material.io/design/components/snackbars.html>.
+  /// This behavior will cause [SnackBar] to be shown on top of other non-fixed
+  /// widgets, such as [FloatingActionButton], rather than pushing it above
+  /// [SnackBar].
+  floating
+}
+
+/// Defines default property values for [SnackBar] widgets.
```

Comment:
Since the SnackBar itself has defined default property values, maybe this should be `Customizes default property values`.

Reply:
done